File: fixtures/scale/scala/large/src/cryptobench/ciphers/ecb/EcbModeSuite.scala
package cryptobench.ciphers.ecb

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class EcbModeSuite extends CipherSuite {
  override def name: String = "ecb"

  override def run(): SuiteResult = {
    val cipher: Cipher = new EcbModeCipher(EcbModeKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- EcbModeVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/ecb/EcbModeSuite.java
package cryptobench.ciphers.ecb;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class EcbModeSuite implements CipherSuite {
    @Override
    public String name() {
        return "ecb";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new EcbModeCipher(EcbModeKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : EcbModeVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}
